File: С++/CWE-476.cpp
#include <iostream>

void dereference_null_pointer() {
    int *ptr = nullptr; // Указатель инициализирован нулем
    std::cout << *ptr << std::endl; // Попытка разыменования нулевого указателя
}

int main() {
    dereference_null_pointer();
    return 0;
}

File: Java/CWE-476.java
public class NullPointerDereference {
    public static void main(String[] args) {
        dereferenceNullPointer();
    }

    public static void dereferenceNullPointer() {
        String str = null; // Инициализация строки нулевым значением
        System.out.println(str.length()); // Попытка вызова метода на нулевой ссылке
    }
}
